### PR TITLE
Fix Kotlin DSL caching soak test

### PR DESCRIPTION
**Pull request by eskatos**
See 
- https://builds.gradle.org/viewLog.html?buildId=26354971&buildTypeId=Gradle_Check_Soak_8_0&tab=buildResultsDiv 
- https://gradle.slack.com/archives/C03A0MT6S/p1567699569132500
- https://github.com/gradle/gradle/pull/10305

### CODE REVIEW

File: subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt [hunk 1]
```diff
@@ -234,19 +243,27 @@ class ScriptCachingIntegrationTest : AbstractScriptCachingIntegrationTest() {
...
 
+        // and: kotlin-dsl cache assertions
+        fun KotlinDslCacheFixture.assertCacheHits(run: Int) {
+            if (run == 1) {
+                misses(settingsFile.stage1)
+                hits(settingsFile.stage2)
+            } else {
+                hits(settingsFile)
+            }
+            hits(settingsFile.stage2)
```
Comment (bamboo) [inline subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt]: ❌ This line can be removed for the condition is already implied by the two branches above.

File: subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt [hunk 2]
```diff
@@ -234,19 +243,27 @@ class ScriptCachingIntegrationTest : AbstractScriptCachingIntegrationTest() {
         val settingsFile = cachedSettingsFile(withSettings(""), false, false)
         val buildFile = cachedBuildFile(withBuildScript("""task<MyTask>("myTask")"""), true)
 
+        // and: kotlin-dsl cache assertions
+        fun KotlinDslCacheFixture.assertCacheHits(run: Int) {
+            if (run == 1) {
```
Comment (bamboo) [inline subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt]: ⭕️ How about adding a comment here about `buildSrc` not affecting the `settingsFile` classpath?

Comment (eskatos) [reply subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt]: Good point, done

File: subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt (hunk 1 above)
Comment (eskatos) [reply subprojects/soak/src/integTest/kotlin/org/gradle/kotlin/dsl/caching/ScriptCachingIntegrationTest.kt]: Ah, I missed that thanks, and done

**Comment by eskatos** [comment]
Soak tests passed on Linux https://builds.gradle.org/viewLog.html?buildId=26466981&buildTypeId=Gradle_Check_Soak_8_0
But failed with an OOME on Windows https://builds.gradle.org/viewLog.html?buildId=26478382&

**Comment by bamboo** [comment]
That OOME probably has the same culprit as the one reported by @blindpirate in https://github.com/gradle/gradle/pull/10414#issuecomment-527292101.

**Pull request merged**